web/src/engine/predictive-text/worker-thread/src/main/correction/context-tracker.ts

```diff
@@ -127,15 +136,7 @@ export class TrackedContextState {
       // Be sure to deep-copy the tokens!  Pointer-aliasing is bad here.
       this.tokens = source.tokens.map(function(token) {
         let copy = new TrackedContextToken();
-        copy.raw = token.raw;
-        copy.replacements = [].concat(token.replacements);
-        copy.activeReplacementId = token.activeReplacementId;
-        copy.transformDistributions = [].concat(token.transformDistributions);
-
-        if(token.replacementText) {
-          copy.replacementText = token.replacementText;
-        }
-
+        Object.assign(copy, token);
```

Comment:
This is not strictly the same because the arrays in copy will be the same instances as in token. Is that okay?

Reply:
Good catch.